src/pages/permissions/panels/PermissionSelector.tsx

```diff
@@ -0,0 +1,187 @@
+import { fetchPermissions } from "api/auth-permissions";
+import { ChangeEvent, FC, useEffect, useState } from "react";
+import { LxdPermission } from "types/permissions";
+import {
+  defaultOption,
+  generateResourceEntitlementOptions,
+  generateResourceOptions,
+  noneAvailableOption,
+  resourceTypeOptions,
+} from "util/permissions";
+import { queryKeys } from "util/queryKeys";
+
+export type LxdPermissionWithID = LxdPermission & { id: string };
+
+interface Props {
+  imageNamesLookup: Record<string, string>;
+  identityNamesLookup: Record<string, string>;
+  onAddPermission: (permission: LxdPermissionWithID) => void;
+}
+
+const PermissionSelector: FC<Props> = ({
+  imageNamesLookup,
+  identityNamesLookup,
+  onAddPermission,
+}) => {
+  const notify = useNotify();
+  const [resourceType, setResourceType] = useState("");
+  const [resource, setResource] = useState("");
+  const [entitlement, setEntitlement] = useState("");
+
+  const {
+    data: permissions,
+    isLoading: isPermissionsLoading,
+    error: permissionsError,
+  } = useQuery({
+    queryKey: [queryKeys.permissions, resourceType],
+    queryFn: () => fetchPermissions({ resourceType }),
+    enabled: !!resourceType,
+  });
+
+  useEffect(() => {
+    document.getElementById("resourceType")?.focus();
+  }, []);
+
+  useEffect(() => {
+    if (!resourceType) {
+      document.getElementById("resourceType")?.focus();
+      return;
+    }
+
+    if (resourceType === "server") {
+      document.getElementById("entitlement")?.focus();
+      return;
+    }
+    document.getElementById("resource")?.focus();
+  }, [resourceType, permissions]);
+
+  useEffect(() => {
+    document.getElementById("entitlement")?.focus();
+  }, [resource]);
+
+  useEffect(() => {
+    document.getElementById("add-entitlement")?.focus();
+  }, [entitlement]);
+
+  const resetSelectors = () => {
+    setResourceType("");
+    setResource("");
+    setEntitlement("");
+  };
+
+  const handleResourceTypeChange = (e: ChangeEvent<HTMLSelectElement>) => {
+    const resourceType = e.target.value;
+    setEntitlement("");
+    setResourceType(resourceType);
+
+    // for server resource type we can select a resource, so automatically set it
+    // label for resource will show "server" if resource type is set to server
+    if (resourceType === "server") {
+      setResource("/1.0");
+    } else {
+      setResource("");
+    }
+  };
+
+  const handleResourceChange = (e: ChangeEvent<HTMLSelectElement>) => {
+    setResource(e.target.value);
+  };
+
+  const handleEntitlementChange = (e: ChangeEvent<HTMLSelectElement>) => {
+    setEntitlement(e.target.value);
+  };
+
+  const handleAddPermission = () => {
+    const newPermissionId = resourceType + resource + entitlement;
+    const isServerPermission = resourceType === "server";
+    const newPermission = {
+      id: newPermissionId,
+      entity_type: resourceType,
+      url: isServerPermission ? "/1.0" : resource,
```

Comment:
The setting of resource state was inconsistent previously because the user couldn't really select a resource for a server permission, I've dealt with that in `handleResourceTypeChange` and forgot to remove this logic here. Will remove the check here, it works fine without it.